### ISSUE TITLE
Generated gradle-api jar contains JRE classes and in old version which confuses Kotlin compiler

### ISSUE THREAD

**Issue by Vampire** [gradle/gradle#16147]
The file `<GRADLE_USER_HOME>/caches/6.8.2/generated-gradle-jars/gradle-api-6.8.2.jar` contains JRE classes.
This majorly confuses the Kotlin compiler which has imho wrong precedence rules and prefers class path classes over JRE classes.
Due to that I for example cannot compile a plugin that uses
```kt
javax.xml.xpath.XPathFactory.newInstance().newXPath().compile("/").evaluateExpression(null)
```
when using the `kotlin-dsl` plugin as Gradle packs in a pre-Java-9 version of the class where the method is missing, even if I compile and run my build with Java 11.

See here for the whole story: https://youtrack.jetbrains.com/issue/KT-44916

While imho this is a clear bug in the Kotlin compiler, maybe the `gradle-api` jar should not package JRE classes.
For Java or Groovy compilation those are not even considered as the JRE classes always win.

**Comment by eskatos** [comment]
This most probably predates the inclusion of `javax.xml` in the JDK ...

With Gradle >= 9.0.0 requiring Java 17 to run and Java 8 for worker processes, all runtimes in use will have the `javax.xml` package built in. We should now be able to safely remove it from the Gradle API JAR.

Related
* https://github.com/gradle/gradle/issues/29483
* https://github.com/gradle/gradle/issues/19577

**Comment by Vampire** [comment]
> This most probably predates the inclusion of javax.xml in the JDK ...

I'm sorry, but I highly doubt that. :-)
Specifically the class `javax.xml.xpath.XPathFactory` I mentioned in in Java since 1.5.
And even Gradle 0.7 had Java 1.5 as minimum Java requirement according to its Userguide. :-)

**Comment by eskatos** [comment]
Right, then this is just a plain mistake to include these in the Gradle API jar.
I suspect this is the case for a very long time.

**Comment by eskatos** [comment]
After some testing, Gradle 8.13 fixed this.

* [8.12 failure scan](https://scans.gradle.com/s/ocjbe6rpyay7e/failures#1)
* [8.13 success scan](https://scans.gradle.com/s/wryk6qovqsxlw)

The following PR adds some test coverage for this:
* https://github.com/gradle/gradle/pull/34307

With this, I'll close this issue as fixed in Gradle 8.13.

----

The Gradle API JAR still contains an old version of `xml-apis:xml-apis` though.
This aspect may be handled as part of the related:
* https://github.com/gradle/gradle/issues/19577

**Comment by eskatos** [comment]
Follow up is here:
* https://github.com/gradle/gradle/pull/34314
